opa.yaml

```
@@ -0,0 +1,84 @@
+package:
+  name: opa
+  version: 0.64.1
+  epoch: 0
+  description: Open Policy Agent (OPA) is an open source, general-purpose policy engine.
+  copyright:
+    - license: Apache-2.0
+
+environment:
+  contents:
+    packages:
+      - build-base
+      - busybox
+      - ca-certificates-bundle
+      - go
+      - wolfi-baselayout
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/open-policy-agent/opa
+      tag: v${{package.version}}
+      expected-commit: 298f97d79da33ede2bd9ec9cbc5641daa63a8587
+
+  - uses: go/build
+    with:
+      ldflags: "-X github.com/open-policy-agent/opa/version.Version=${{package.version}}"
+      modroot: .
+      packages: .
+      output: opa
+
+  - uses: strip
+
+  - runs: |
+      mkdir -p ${{targets.contextdir}}/opa
+
+      cat << EOF > ${{targets.contextdir}}/opa/example.rego
+      package authz
+      import rego.v1
+
+      allow if {
+        input.path == ["users"]
+        input.method == "POST"
+      }
+
+      allow if {
+        input.path == ["users", input.user_id]
+        input.method == "GET"
+      }
+      EOF
+
+      cat << EOF > ${{targets.contextdir}}/opa/example_test.rego
+      package authz
+      import rego.v1
+
+      test_post_allowed if {
+        allow with input as {"path": ["users"], "method": "POST"}
+      }
+
+      test_get_anonymous_denied if {
+        not allow with input as {"path": ["users"], "method": "GET"}
+      }
+
+      test_get_user_allowed if {
+        allow with input as {"path": ["users", "bob"], "method": "GET", "user_id": "bob"}
+      }
+
+      test_get_another_user_denied if {
+        not allow with input as {"path": ["users", "bob"], "method": "GET", "user_id": "alice"}
+      }
+      EOF
+
+update:
+  enabled: true
+  github:
+    identifier: open-policy-agent/opa
+    strip-prefix: v
+    use-tag: true
+
+test:
+  pipeline:
+    - runs: |
+        opa version
+        cd /opa && opa test . -v
```

```suggestion
        opa test . -v
```

If the example files live in the `opa` directory here in the repo, then they will be available in the test pipeline. Here's an example showing them being invoked:

```
2024/05/02 15:52:48 WARN + opa test . -v
2024/05/02 15:52:48 INFO example_test.rego:
2024/05/02 15:52:48 INFO data.authz.test_post_allowed: PASS (374.173µs)
2024/05/02 15:52:48 INFO data.authz.test_get_anonymous_denied: PASS (157.69µs)
2024/05/02 15:52:48 INFO data.authz.test_get_user_allowed: PASS (215.944µs)
2024/05/02 15:52:48 INFO data.authz.test_get_another_user_denied: PASS (166.987µs)
2024/05/02 15:52:48 INFO --------------------------------------------------------------------------------
2024/05/02 15:52:48 INFO PASS: 4/4
```